src/main/java/org/sagebionetworks/web/client/presenter/EntityPresenter.java

```diff
@@ -320,14 +322,22 @@ public Widget asWidget() {
 
   @EventHandler
   public void onEntityUpdatedEvent(EntityUpdatedEvent event) {
-    List<SynapseReactClientQueryKey> queryKey = SynapseReactClientQueryKey.create(
+    List<SynapseReactClientEntityQueryKey> queryKey = SynapseReactClientEntityQueryKey.create(
       "entity",
       event.getEntityId()
     );
     queryClient.resetQueries(queryKey);
     globalAppState.refreshPage();
   }
 
+  @EventHandler
+  public void onDownloadListUpdatedUpdatedEvent(
+    DownloadListUpdatedEvent _event
+  ) {
+    List<String> queryKey = Collections.singletonList("downloadList");
+    queryClient.invalidateQueries(queryKey);
+  }
```

Comment:
Invalidate the `downloadList` cache items when the download list is updated. Fixes an issue where the count and download list won't immediately update after adding an item to the download cart.

Reply:
https://sagebionetworks.jira.com/browse/SWC-6353